### PR TITLE
Initial POC of RabbitMQ transport

### PR DESCRIPTION
**Pull request by devkits**
### What does this PR do?
Initial POC implementation of RabbitMQ as a new transport layer.
(work-in-progress, **but looking for early feedback before the review gets too large**).

  - there are a number of TODOs in this review; all should be addressed before review is ready for prime time
  - there are gaps in the implementation that will be addressed. This work is tracked in SSC-978
  - a number of edge cases need to be tested/addressed (e.g. connection error recovery, etc.)
  - added functional tests
  - added integration tests

Testing Done: functional tests with on master and one minion

### What issues does this PR fix or reference?
Fixes: VMware SSC-978

### Merge requirements satisfied?
**[NOTICE] Bug fixes or features added to Salt require tests.**

- [ ] Docs
- [ ] Changelog - https://docs.saltproject.io/en/master/topics/development/changelog.html
- [X] Tests written/updated

### Commits signed with GPG?
No

### CODE REVIEW

File: tests/pytests/functional/transport/zeromq/test_pub_server_channel.py [hunk 1]
```diff
@@ -204,13 +204,13 @@ def __exit__(self, *args):
...
     """
-    Test sending 10K messags to ZeroMQPubServerChannel using IPC transport
+    Test sending 10K messages to ZeroMQPubServerChannel using IPC transport
 
     ZMQ's ipc transport not supported on Windows
     """
     opts = dict(salt_master.config.copy(), ipc_mode="ipc", pub_hwm=0)
     with PubServerChannelProcess(opts, salt_minion.config.copy()) as server_channel:
-        send_num = 10000
+        send_num = 4
```
Comment (devkits) [inline tests/pytests/functional/transport/zeromq/test_pub_server_channel.py:213]: I should undo this change. This number was reduced to easier testing and comparison of rabbitmq and zeromq.

**Comment by devkits** [comment]
Closing in favour of https://github.com/saltstack/salt/pull/60775
